tests-e2e/core/applicationsnapshotenvironmentbinding_test.go

```diff
@@ -431,6 +431,130 @@ var _ = Describe("ApplicationSnapshotEnvironmentBinding Reconciler E2E tests", f
+
+			By("Verify that Status.GitOpsDeployments field of Binding is having Component and GitOpsDeployment name.")
+			gitOpsDeploymentName := appstudiocontroller.GenerateBindingGitOpsDeploymentName(binding, binding.Spec.Components[0].Name)
+
+			expectedGitOpsDeployments := []appstudiosharedv1.BindingStatusGitOpsDeployment{{
+				ComponentName: binding.Spec.Components[0].Name, GitOpsDeployment: gitOpsDeploymentName,
+			}}
+
+			Eventually(binding, "2m", "1s").Should(bindingFixture.HaveStatusGitOpsDeployments(expectedGitOpsDeployments))
+
+			By("Verify whether `gitopsDeployment.ObjectMeta.Labels` is updated with ASEB labels")
+			gitopsDeployment := managedgitopsv1alpha1.GitOpsDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      gitOpsDeploymentName,
+					Namespace: binding.Namespace,
+				},
+			}
+
+			err = k8s.Get(&gitopsDeployment)
+			Expect(err).To(BeNil())
+			Expect(gitopsDeployment.ObjectMeta.Labels).ToNot(BeNil())
+			Expect(gitopsDeployment.ObjectMeta.Labels).To(Equal(map[string]string{"appstudio.openshift.io": "testing"}))
+		})
+
+		It("Should not append ASEB label without prefix appstudio.openshift.io into the GitopsDeployment Label", func() {
+			By("Create ApplicationSnapshotEnvironmentBindingResource")
+			binding := buildApplicationSnapshotEnvironmentBindingResource("appa-staging-binding", "new-demo-app", "staging", "my-snapshot", 3, []string{"component-a"})
+			err := k8s.Create(&binding)
+			Expect(err).To(Succeed())
+
+			By("Update Status field of ApplicationSnapshotEnvironmentBindingResource")
+			err = k8s.Get(&binding)
+			Expect(err).To(Succeed())
+			binding.Status = buildApplicationSnapshotEnvironmentBindingStatus(binding.Spec.Components, "https://github.com/redhat-appstudio/gitops-repository-template", "main", "fdhyqtw", []string{"components/componentA/overlays/staging"})
+			err = k8s.UpdateStatus(&binding)
+			Expect(err).To(Succeed())
+
+			By("Verify that Status.GitOpsDeployments field of Binding is having Component and GitOpsDeployment name.")
+			gitOpsDeploymentName := appstudiocontroller.GenerateBindingGitOpsDeploymentName(binding, binding.Spec.Components[0].Name)
+
+			expectedGitOpsDeployments := []appstudiosharedv1.BindingStatusGitOpsDeployment{{
+				ComponentName: binding.Spec.Components[0].Name, GitOpsDeployment: gitOpsDeploymentName,
+			}}
+
+			Eventually(binding, "2m", "1s").Should(bindingFixture.HaveStatusGitOpsDeployments(expectedGitOpsDeployments))
+
+			By("Verify whether `gitopsDeployment.ObjectMeta.Labels` is not updated with ASEB labels")
+			gitopsDeployment := managedgitopsv1alpha1.GitOpsDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      gitOpsDeploymentName,
+					Namespace: binding.Namespace,
+				},
+			}
+
+			err = k8s.Get(&gitopsDeployment)
+			Expect(err).To(BeNil())
+			Expect(gitopsDeployment.ObjectMeta.Labels).To(BeNil())
+			Expect(gitopsDeployment.ObjectMeta.Labels).ToNot(Equal(map[string]string{"appstudio.openshift.io": "testing"}))
```

Comment:
I guess in this case map will be empty, if not, then map should not have the entry with given key, you should check that as well.

Reply:
Map will be empty here @jparsai